GeneratorInterface/Core/src/WeightHelper.cc

```diff
@@ -0,0 +1,335 @@
+#include "GeneratorInterface/Core/interface/WeightHelper.h"
+
+#include <regex>
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Utilities/interface/Exception.h"
+
+namespace gen {
+  WeightHelper::WeightHelper() {}
+
+  bool WeightHelper::isScaleWeightGroup(const ParsedWeight& weight) const {
+    return (weight.groupname.find("scale_variation") != std::string::npos ||
+            weight.groupname.find("Central scale variation") != std::string::npos);
+  }
+
+  bool WeightHelper::isPdfWeightGroup(const ParsedWeight& weight) const {
+    const std::string& name = weight.groupname;
+    if (name.find("PDF_variation") != std::string::npos)
+      return true;
+    return LHAPDF::lookupLHAPDFID(name) != -1;
+  }
+
+  bool WeightHelper::isPartonShowerWeightGroup(const ParsedWeight& weight) const {
+    const std::string& groupname = boost::to_lower_copy(weight.groupname);
+    std::vector<std::string> psNames = {"isr", "fsr", "nominal", "baseline", "emission"};
+    for (const auto& name : psNames) {
+      if (groupname.find(name) != std::string::npos)
+        return true;
+    }
+    return false;
+  }
+
+  bool WeightHelper::isOrphanPdfWeightGroup(ParsedWeight& weight) const {
+    std::pair<std::string, int> pairLHA;
+    try {
+      pairLHA = LHAPDF::lookupPDF(stoi(searchAttributes("pdf", weight)));
+    } catch (...) {
```

Comment:
CMSSW code rules check warns about `catch`
```bash
Rule3 : https://raw.githubusercontent.com/cms-sw/cmssw/master/Utilities/ReleaseScripts/python/cmsCodeRules/config.py
Search for "catch(...)" statements in *.cc, *.cxx files

/GeneratorInterface/Core/src/WeightHelper.cc
[37]
/GeneratorInterface/LHEInterface/plugins/ExternalLHEProducer.cc
[274]
```

In general, do not catch exceptions – leave them to the Framework (see [Exception Guidelines](https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideEdmExceptionUse#Exception%20Handling))

Reply:
In this case, we want execution to continue after an exception may be issued by an external package (LHAPDF), so I think catching is appropriate. I think you can put `CMS_SA_ALLOW` before `try` to suppress the warning.